src/iOS/Xamarin.iOS/Samples/GeometryEngine/Buffer/Buffer.cs

```diff
@@ -48,124 +57,245 @@ public override void ViewDidLoad()
+                LineBreakMode = UILineBreakMode.WordWrap,
+                Lines = 0
+            };
+
+            // Create the buffer input area (toolbar).
+            _bufferInputArea = new UIToolbar();
+
+            // Create the buffer input label.
+            _bufferInputLabel = new UILabel
+            {
+                TextColor = View.TintColor,
+                Text = "Distance (miles):",
+                TextAlignment = UITextAlignment.Left
+            };
+
+            // Create the buffer input control.
+            _bufferDistanceMilesTextField = new UITextField
+            {
+                BackgroundColor = UIColor.White,
+                KeyboardType = UIKeyboardType.NumberPad,
+                Text = "1000",
+                TextAlignment = UITextAlignment.Right,
+                TextColor = View.TintColor
+            };
+            _bufferDistanceMilesTextField.Layer.CornerRadius = 5;
+            
+            // Allow pressing 'return' to dismiss the keyboard.
+            _bufferDistanceMilesTextField.ShouldReturn += (textField) => { textField.ResignFirstResponder(); return true; };
+
+            // Create the legend toolbar.
+            _legendArea = new UIToolbar();
+
+            // Create the label to show the planar buffer color.
+            _planarSwatchLabel = new UILabel
+            {
+                AdjustsFontSizeToFitWidth = true,
+                TextColor = UIColor.White,
+                Text = "Planar Buffers",
+                TextAlignment = UITextAlignment.Center
+            };
+
+            // Create the label to show the geodesic buffer color.
+            _geodesicSwatchLabel = new UILabel
+            {
+                AdjustsFontSizeToFitWidth = true,
+                TextColor = UIColor.White,
+                Text = "Geodesic Buffers",
+                TextAlignment = UITextAlignment.Center
+            };
+
+            // Create the clear buffers button.
+            _clearBuffersButton = new UIButton
+            {
+                ClipsToBounds = true,
+                BackgroundColor = View.TintColor
+            };
+            _clearBuffersButton.SetTitle("Clear", UIControlState.Normal);
+            _clearBuffersButton.SetTitleColor(UIColor.White, UIControlState.Normal);
+            _clearBuffersButton.Layer.CornerRadius = 10;
+
+            // Handle the clear buffers button press.
+            _clearBuffersButton.TouchUpInside += ClearBuffersButton_TouchUpInside; ;
+
+            // Add views to the page.
+            View.AddSubviews(_myMapView,
+                _helpLabel,
+                _bufferInputLabel,
+                _bufferDistanceMilesTextField,
+                _legendArea,
+                _planarSwatchLabel,
+                _geodesicSwatchLabel,
+                _clearBuffersButton);
+        }
+
+        public override void ViewDidLayoutSubviews()
+        {
+            var topMargin = NavigationController.NavigationBar.Frame.Height +
```

Comment:
This hasn't been shared broadly with the team yet, but there is a problem where sometimes when navigating between samples, this method will be called after the view is disassociated with its controller. This causes a NullReferenceException crash. 

I recommend wrapping the body of this method in a try/catch to avoid that problem. In the future we'll find a better way (probably auto layout).